libs/ui/src/lib/Typography/index.tsx

```diff
@@ -0,0 +1,94 @@
+import { type VariantProps, cva } from "class-variance-authority";
+
+export enum TypographyVariantEnum {
+  Heading = "heading",
+  Heading1 = "heading1",
+  Body = "body",
+  Body1 = "body1",
+  Body2 = "body2",
+  Link = "link",
+}
+
+type JSXElement =
```

Comment:
Non-blocking: There is an in-built type for this! `import type { ElementType } from "react";
` if I remember correctly. You can pass in the specific tag using TypeScript generics.